azure/services/aso/service.go

```diff
@@ -31,6 +32,9 @@ type Service[T deepCopier[T], S Scope] struct {
 
 	Scope S
 	Specs []azure.ASOResourceSpecGetter[T]
+	// ListFunc is used to enumerate ASO resources to determine which stale ones to delete. This behavior may
+	// be skipped for a service by leaving this field nil.
```

Comment:
I elected to leave this unspecified for services like managedclusters and agentpools where I think we can be confident that a 1:1 mapping of CAPZ resource to ASO resource will always exist and we'd never delete an ASO ManagedCluster without deleting the AzureManagedControlPlane.